pkg/graveler/ref/manager.go

```diff
@@ -44,6 +36,15 @@ type Manager struct {
 	batchExecutor   batch.Batcher
 	repoCache       cache.Cache
 	commitCache     cache.Cache
+
+	// MaxBatchDelay - 3ms was chosen as a max delay time for critical path queries.
+	// It trades off amount of queries per second (and thus effectiveness of the batching mechanism) with added latency.
+	// Since reducing # of expensive operations is only beneficial when there are a lot of concurrent requests,
+	//
+	//	the sweet spot is probably between 1-5 milliseconds (representing 200-1000 requests/second to the data store).
+	//
+	// 3ms of delay with ~300 requests/second per resource sounds like a reasonable tradeoff.
```

Comment:
This should probably be near the default value now, as it justifies picking that value.